src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlImpl.java

```diff
@@ -382,4 +451,97 @@ public void destroy() {
+            this.addControlHandleConsumer = addControlHandleConsumer;
+            this.point = point;
+        }
+
+        @Override
+        public void run() {
+            addControlHandleConsumer.accept(point);
+        }
+    }
+
+    @Override
+    public Shape<?> createTransientControlHandle(final Consumer<Point2D> addControlHandleConsumer) {
+        final Shape<?> pointHandleShape = new Circle(R0);
+        m_pointHandleDecorator.decorate(pointHandleShape, IShapeDecorator.ShapeState.INVALID);
+        //increase the selection area to make it easier do drag
+        pointHandleShape.getAttributes().setSelectionBoundsOffset(SELECTION_OFFSET);
+        pointHandleShape.getAttributes().setSelectionStrokeOffset(SELECTION_OFFSET);
+        pointHandleShape.setFillBoundsForSelection(true);
+
+        //adding the handlers on the transient control handle
+        transientControlHandleRegistrations.add(pointHandleShape.addNodeMouseDownHandler(new NodeMouseDownHandler() {
+            @Override
+            public void onNodeMouseDown(NodeMouseDownEvent event) {
+                if(addControlPointTimer == null){
+                    addControlPointTimer = createAddControlPointTimer(new Point2D(event.getX(), event.getY()), addControlHandleConsumer);
+                    addControlPointTimer.schedule(150);
+                    event.preventDefault();
+                }
+            }
+        }));
+
+        transientControlHandleRegistrations.add(pointHandleShape.addNodeMouseClickHandler(new NodeMouseClickHandler() {
+            @Override
+            public void onNodeMouseClick(NodeMouseClickEvent event) {
+                if(addControlPointTimer != null){
+                    //when clicking then cancel the add connector action and fire the click on the connector line
+                    addControlPointTimer.cancel();
+                    addControlPointTimer = null;
+                    event.preventDefault();
+                    m_connector.getLine().fireEvent(event);
+                }
+            }
+        }));
+
+        //add the shape on the connector line
+        m_connector.getLine().getLayer().add(pointHandleShape);
+
+        this.transientControlHandle = pointHandleShape;
+        return pointHandleShape;
+    }
+
+    protected AddControlPointTimer createAddControlPointTimer(Point2D point, Consumer<Point2D> addControlHandleConsumer) {
+        return new AddControlPointTimer(addControlHandleConsumer, point);
+    }
+
+    @Override
+    public void destroyTransientControlHandle() {
+        for (HandlerRegistration registration : transientControlHandleRegistrations) {
+            registration.removeHandler();
+        }
+        transientControlHandleRegistrations.clear();
+        if (transientControlHandle != null) {
+            transientControlHandle.removeFromParent();
+            transientControlHandle = null;
+            batchConnector();
+        }
+        addControlPointTimer = null;
```

Comment:
I'm not expert here, so it's more a question than an objection. Is it possible to reach this point when the timer is "alive", and thus a timer cancel could be required?